lib/rucio/client/uploadclient.py

```diff
@@ -625,8 +626,15 @@ def _upload_item(self, rse_settings, rse_attributes, lfn, source_dir=None, domai
         pfn_tmp = '%s.rucio.upload' % pfn if protocol_write.renaming else pfn
 
         # Either DID eixsts or not register_after_upload
-        if protocol_write.overwrite is False and delete_existing is False and protocol_write.exists(pfn):
-            raise FileReplicaAlreadyExists('File %s in scope %s already exists on storage as PFN %s' % (name, scope, pfn))  # wrong exception ?
+        if protocol_write.overwrite is False and delete_existing is False:
+            if sign_service:
+                # Construct protocol for read ONLY for cloud resources and get signed URL for GET
+                protocol_read = self._create_protocol(rse_settings, 'read', domain=domain, impl=impl)
+                readpfn = self.client.get_signed_url(rse_settings['rse'], sign_service, 'read', pfn)    # NOQA pylint: disable=undefined-variable
+                if protocol_read.exists(readpfn):
+                    raise FileReplicaAlreadyExists('File %s in scope %s already exists on storage as PFN %s' % (name, scope, pfn))  # wrong exception ?
+            if not sign_service and protocol_write.exists(pfn):
```

Comment:
lgtm, just one question, what happens when not sign_service and not protocol_write.exists(pfn)?